providers/dns/godaddy/client.go

```diff
@@ -7,35 +7,64 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
+	"path"
 )
 
 // DNSRecord a DNS record
 type DNSRecord struct {
-	Type     string `json:"type"`
-	Name     string `json:"name"`
+	Name     string `json:"name,omitempty"`
+	Type     string `json:"type,omitempty"`
 	Data     string `json:"data"`
 	Priority int    `json:"priority,omitempty"`
 	TTL      int    `json:"ttl,omitempty"`
 }
 
-func (d *DNSProvider) updateRecords(records []DNSRecord, domainZone string, recordName string) error {
+func (d *DNSProvider) getRecords(domainZone string, rType string, recordName string) ([]DNSRecord, error) {
+	resource := path.Clean(fmt.Sprintf("/v1/domains/%s/records/%s/%s", domainZone, rType, recordName))
+
+	resp, err := d.makeRequest(http.MethodGet, resource, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() { _ = resp.Body.Close() }()
```

Comment:
This doesn't have any functional difference from:

```go
defer resp.Body.Close()
```

Is this because of linting?

Reply:
Currently, I prefer `defer func() { _ = something.Close() }()` due to the explicit ignore of the error.
It's a bit more verbose but it's mush easy to read and check ignored errors.

Reply:
As a general rule of thumb, if that's the direction that lego should be moving, then I'd suggest seeing if it can be added as a linting rule. Specifically to ensure consistency throughout the rest of the codebase in the future. Just a recommendation!